backend/src/main/java/de/neuefische/backend/service/EventHubUserDetailService.java

```diff
@@ -36,14 +41,38 @@ public EventHubUserDTO saveUser(UserLoginAndRegistrationDTO user) {
                 .id(generateUUIDService.generateUUID())
                 .username(user.getUsername())
                 .password(passwordEncoder.encode(user.getPassword()))
+                .preferredCategories(new ArrayList<>())
                 .build();
 
-        if(userRepository.findEventHubUserByUsername(newUser.getUsername()).isPresent()) {
+        if (userRepository.findEventHubUserByUsername(newUser.getUsername()).isPresent()) {
             throw new IllegalArgumentException("User with username " + user.getUsername() + " already exists");
         }
 
         EventHubUser temp = userRepository.save(newUser.withRoles(Collections.singletonList(new SimpleGrantedAuthority("user"))).withPassword(passwordEncoder.encode(user.getPassword())));
         List<String> roles = temp.getRoles().stream().map(SimpleGrantedAuthority::toString).toList();
-        return new EventHubUserDTO(temp.getId(), temp.getUsername(), roles);
+        return new EventHubUserDTO(temp.getId(), temp.getUsername(), roles, temp.getPreferredCategories());
     }
+
+    public List<String> getUserPreferredCategories(String username) {
+        EventHubUser user = userRepository.findEventHubUserByUsername(username)
+                .orElseThrow(() -> new UsernameNotFoundException(String.format(USER_NOT_FOUND_MESSAGE, username)));
+
+        List<EventCategory> preferredCategories = user.getPreferredCategories();
+
+        if (preferredCategories == null) {
+            return Collections.emptyList();
+        }
+
+        return user.getPreferredCategories().stream()
```

Comment:
You define preferedCategories but you don't use it more then once. Why not if pref.Categories == null ... else return ... ? Why save it in preferredCategories at all?